### ISSUE TITLE
Cutting releases on a release branch

### ISSUE THREAD

**Issue by marten-seemann** [protocol/.github#229]
We only run the `releaser` workflow on the default branch: https://github.com/protocol/.github/blob/9c84fce2b8c4386e4e7558830825298c207e74ec/templates/.github/workflows/releaser.yml#L1-L5

That means it's not possible to cut a release on a release branch. I'm not sure what the right solution is. We definitely don't want to cut the release on every `push` to any branch, this would make it impossible to create a PR in the first place.

cc @mvdan @galargh

**Comment by mvdan** [comment]
As an idea, you could limit it to branches named `main`, `master`, and `release-.*`.

**Comment by marten-seemann** [comment]
`release-.*` is dangerous. I have created branches named `release-v0.4.0` (as a PR to master) before.

`release-branch-.*` or `off-master-release-.*` might be less prone to accidental releases.

**Comment by marten-seemann** [comment]
I really wish there was a way to get the PR when `push` is run (if the `push` event was triggered by a PR merge).
Then we could for example introduce a `release` label: Only if this label is set on the PR, will a merge to a branch (other than `master`) trigger a release.

**Comment by marten-seemann** [comment]
Turns out there's an Action for that: https://github.com/actions-ecosystem/action-get-merged-pull-request.

We can distinguish between a normal `push` and a `push` that resulted from a PR being merged by looking at the output of this action, for example by checking if `output.title == ''`:
```yml
 - uses: actions-ecosystem/action-get-merged-pull-request@v1
    id: getmergedpr
    with:
      github_token: ${{ secrets.GITHUB_TOKEN }}
  - run: <do stuff if a PR was merged>
    if: steps.getmergedpr.outputs.title != ''
```

This works for PR resulting from a merge commit and also for a squash merge (probably also for a rebase merge, but I haven't checked that).

**Comment by marten-seemann** [comment]
Fixed by #230.
